gammapy/image/models/core.py

```diff
@@ -170,6 +172,136 @@ def evaluate(lon, lat, lon_0, lat_0, r_0):
+        rc('font',**{'family':'sans-serif','sans-serif':['Helvetica']})
+        ## for Palatino and other serif fonts use:
+        #rc('font',**{'family':'serif','serif':['Palatino']})
+        rc('text', usetex=True)
+
+        ell=SkyEllipse(2*u.deg,2*u.deg,1*u.deg,.8,30*u.deg)
+
+        m_geom = WcsGeom.create(binsz=.01,width=(3,3),skydir=(2,2),coordsys="GAL",proj="AIT")
+        coords=m_geom.get_coord()
+        lon=coords.lon*u.deg
+        lat=coords.lat*u.deg  
+        vals=ell(lon,lat)
+        mymap=Map.from_geom(m_geom,data=vals.value)
+
+        fig,ax,_=mymap.plot()
+        ax.scatter(2, 2, transform=ax.get_transform('galactic'), s=20,edgecolor='red', facecolor='red')
+        plt.text(2.08, 2.06, r'$(l_0,b_0)$' ,transform=ax.get_transform('galactic'), fontsize=12)
+        plt.plot([2,2+np.cos(np.pi/6)],[2,2+np.sin(np.pi/6)], transform=ax.get_transform('galactic'))
+        ax.hlines(y=2,color='r', linestyle='--',transform=ax.get_transform('galactic'),xmin=0,xmax=5)
+        plt.text(2.4, 2.06, r'$\theta$' ,transform=ax.get_transform('galactic'), fontsize=12)
+
+        plt.show()
+    """
+
+    def __init__(self, lon_0, lat_0, semi_major, e, theta):
+        self.parameters = Parameters(
+            [
+                Parameter("lon_0", Longitude(lon_0)),
+                Parameter("lat_0", Latitude(lat_0)),
+                Parameter("semi_major", Angle(semi_major)),
+                Parameter("e", e, min=0, max=1),
+                Parameter("theta", Angle(theta)),
+            ]
+        )
+
+    def find_foci(lon_0, lat_0, semi_major, e, theta):
+        """Find the foci of the ellipse."""
+        c = semi_major * e
+        lon_1, lat_1 = offset_by(lon_0, lat_0, 90 * u.deg - theta, c)
+        lon_2, lat_2 = offset_by(lon_0, lat_0, 270 * u.deg - theta, c)
+        return lon_1, lat_1, lon_2, lat_2
+
+    def compute_norm(semi_major, e):
+        """Compute the normalization factor."""
+        semi_minor = semi_major * np.sqrt(1 - e ** 2)
+
+        def integral_fcn(x, a, b):
+            A = 1 + 1 / a.to("rad").value ** 2
+            B = 1 + 1 / b.to("rad").value ** 2
+            C = A - B
+            cs2 = np.cos(x) ** 2
+            return 1 - np.sqrt(1 - 1 / (B + C * cs2))
+
+        return (
+            quad(lambda x: integral_fcn(x, semi_major, semi_minor), 0, 2 * np.pi)[0]
+            ** -1
+        )
+
+    @staticmethod
+    def evaluate(lon, lat, lon_0, lat_0, semi_major, e, theta):
+        """Evaluate the model (static function)."""
+        lon_1, lat_1, lon_2, lat_2 = SkyEllipse.find_foci(
+            lon_0, lat_0, semi_major, e, theta
+        )
+        sep_1 = angular_separation(lon, lat, lon_1, lat_1)
+        sep_2 = angular_separation(lon, lat, lon_2, lat_2)
+        in_ellipse = sep_1 + sep_2 <= 2 * semi_major
+        norm = u.Quantity(
+            SkyEllipse.compute_norm(semi_major, e), unit="sr-1", copy=False
+        )
+        result = np.select([in_ellipse], [norm])
```

Comment:
I think `np.select()` does not work properly with `Quantity`, that's probably why the instance check is needed later. You could simplify the code as following:

```
norm = SkyEllipse.compute_norm()
return u.Quantity(norm * in_ellipse, "sr-1", copy=False)
```
This implicitly does the type-cast of the bool array `in_ellipse`, but it works correctly with `Quantity` as well.